### PR TITLE
feat(mobile): thumbhash on mobile

### PR DESCRIPTION
**Pull request by shenlong-tanwen**
Continues on #2913 

#### Changes made in the PR

- Caught up to main and updated asset grid view to use thumbhash on scroll as well and not just when rendering the image
- Wraps `ThumbnailImage` in asset grid view with `RepaintBoundary` to prevent repaint of other assets in the same row due to repaint of a different asset

> [!Note]
> I did not test it extensively and not sure about the performance implications of loading the assets even during scrolling. The flickering is not completely removed though, but it is vastly reduced in my initial level of testing

### CODE REVIEW

File: mobile/lib/modules/home/ui/asset_grid/immich_asset_grid_view.dart [hunk 1]
```diff
@@ -254,9 +264,8 @@ class ImmichAssetGridViewState extends State<ImmichAssetGridView> {
             widget.margin * (widget.assetsPerRow - 1) / widget.assetsPerRow;
         final rows =
             (section.count + widget.assetsPerRow - 1) ~/ widget.assetsPerRow;
-        final List<Asset> assetsToRender = scrolling
-            ? []
-            : widget.renderList.loadAssets(section.offset, section.count);
+        final List<Asset> assetsToRender =
```
Comment (fyfrey) [inline mobile/lib/modules/home/ui/asset_grid/immich_asset_grid_view.dart:267]: This loads many assets quickly and often from DB while scrolling. We'd need to test the performance implications for large libraries (500k assets) on medium-powered devices.

Comment (shenlong-tanwen) [reply mobile/lib/modules/home/ui/asset_grid/immich_asset_grid_view.dart:267]: Yes. This does load the assets for each of the section during scrolling. I gave it a test only in the simulator using the demo instance which is not much. This change is to also show the thumb hash on scrolling with the scroll bar.

The main change that vastly reduced the flickering was using the `RepaintBoundary` around `ThumbnailImage`. I tested it after removing all the changes I made from `immich_asset_grid_view` and except wrapping the thumbnails with repaintboundary.

We can remove the thumbhash on scroll change if the performance is bad on low - medium end devices. I don't have that many assets to test 😅  Would be great if someone could test it

**Comment by alextran1502** [comment]
Hello, I think this works a lot better now. One scenario I noticed is that if we keep holding the thumb scroll and not release it, it will keep the section rebuilt. I think we can optimize it by disable thumbhash loading unless the thumb scroll is released

**Comment by shenlong-tanwen** [comment]
> Hello, I think this works a lot better now. One scenario I noticed is that if we keep holding the thumb scroll and not release it, it will keep the section rebuilt. I think we can optimize it by disabling thumbhash loading unless the thumb scroll is released.
> 
> I think it will also reduce the "jitter" when dragging the thumb scroll in a large album view

Yes, we can do that. Fynn also had concerns regarding the performance of loading the assets from db on scroll which is needed to display the thumbhash on thumb scroll. Should I remove the scrolling behavior with thumb scroll back to displaying gray placeholders?

**Comment by alextran1502** [comment]
> > Hello, I think this works a lot better now. One scenario I noticed is that if we keep holding the thumb scroll and not release it, it will keep the section rebuilt. I think we can optimize it by disabling thumbhash loading unless the thumb scroll is released.
> > I think it will also reduce the "jitter" when dragging the thumb scroll in a large album view
> 
> Yes, we can do that. Fynn also had concerns regarding the performance of loading the assets from db on scroll which is needed to display the thumbhash on thumb scroll. Should I remove the scrolling behavior with thumb scroll back to displaying gray placeholders?

Yes, I think this would be preferred. We don't want to sacrifice performance for the aesthetic look in this use case 😄 .

**Comment by shenlong-tanwen** [comment]
> Yes, I think this would be preferred. We don't want to sacrifice performance for the aesthetic look in this use case 😄 .

Updated my change now to only add a `RepaintBoundary` around the `ThumbnailImage` and removed the thumbhash on scroll 😅

**Comment by shenlong-tanwen** [comment]
Will re-implement this on top of #7016

**Pull request closed without merging**